do/csi/driver/mounter.go

```diff
@@ -0,0 +1,107 @@
+package driver
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// Mounter is responsible for formatting and mounting volumes
+type Mounter interface {
+	// Format formats the source with the given filesystem type
+	Format(source, fsType string) error
+
+	// Mount mounts source to target with the given fstype and options.
+	Mount(source, target, fsType string, options ...string) error
+
+	// Unmount unmounts the given target
+	Unmount(target string) error
+}
+
+// TODO(arslan): this is Linux only for now. Refactor this into a package with
+// architecture specific code in the future, such as mounter_darwin.go,
+// mounter_linux.go, etc..
+type mounter struct{}
+
+func (m *mounter) Format(source, fsType string) error {
+	mkfsCmd := fmt.Sprintf("mkfs.%s", fsType)
+	mkfsArgs := []string{}
+
+	if fsType == "" {
+		return errors.New("fs type is not specified for formatting the volume")
+	}
+
+	if source == "" {
+		return errors.New("source is not specified for formatting the volume")
+	}
+
+	mkfsArgs = append(mkfsArgs, source)
+	if fsType == "ext4" || fsType == "ext3" {
+		mkfsArgs = []string{"-F", source}
+	}
+
+	out, err := exec.Command(mkfsCmd, mkfsArgs...).CombinedOutput()
```

Comment:
check if `mkfs.%s` is present with https://golang.org/pkg/os/exec/#LookPath prior executing

Reply:
good call, fixed with: https://github.com/digitalocean/digitalocean-cloud-controller-manager/pull/77/commits/f763ca9bf39d0a92f72512f3feca116de709b20b

Reply:
nit. Might be a good idea to use the path returned by `LookPath` for the mkfsCmd. Not necessary